utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -630,35 +712,193 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
 
         debug_assert!(have_value);
 
+        // Now that c >= high_start, compare `value` to `high_value` to see
+        // if we can extend our current range to CODE_POINT_MAX or stop at
+        // high_start - 1.
         let di: u32 = self.data.len() as u32 - HIGH_VALUE_NEG_DATA_OFFSET;
-        let high_value: u32 = if let Some(hv) = self.data.get(di as usize) {
-            hv.into()
-        } else {
-            return None;
-        };
-        if maybe_filter_value(high_value, self.header.null_value, null_value) != value {
+        let high_value: T = self.data.get(di as usize)?;
+        if maybe_filter_value(
+            high_value,
+            T::try_from_u32(self.header.null_value).ok()?,
+            null_value,
+        ) != value
+        {
             c -= 1;
         } else {
             c = CODE_POINT_MAX;
         }
         Some(CodePointMapRange {
-            start,
-            end: c,
+            range: RangeInclusive::new(start, c),
             value,
         })
     }
 
     /// Yields an [`Iterator`] returning ranges of consecutive code points that
-    /// share the same value in the [`CodePointTrie`].
-    pub fn iter_ranges(&'trie self) -> CodePointMapRangeIterator<'trie, T> {
+    /// share the same value in the [`CodePointTrie`], as given by
+    /// [`CodePointTrie::get_range()`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use core::ops::RangeInclusive;
+    /// use icu_codepointtrie::CodePointMapRange;
+    /// use icu_codepointtrie::planes;
+    ///
+    /// let planes_trie = planes::get_planes_trie();
+    ///
+    ///
+    /// let mut ranges = planes_trie.iter_ranges();
+    ///
+    /// assert_eq!(
+    ///     ranges.next(),
+    ///     Some(CodePointMapRange {
+    ///         range: RangeInclusive::new(0x0, 0xffff),
+    ///         value: 0
+    ///     })
+    /// );
+    ///
+    /// assert_eq!(
```

Comment:
Suggestion: I like this example, but it is lengthy.  I think examples are best when they are not as many lines of code.  Can you refactor this into a loop of some sort?  Maybe iterate *i* from 0 to 16 and check the value being *i* and the start code point being *i \* 0x10000*

Reply:
Done. That sounds better. I misunderstood your intent from your wording when you wanted the previous `.skip(14)` to be replaced with checks against the each of the returned range values.